### DISCUSSION TITLE
Socket client ip address

### DISCUSSION THREAD

**Discussion by shishamo** [Totodore/socketioxide#377] [Q&A] [answered]
Hello there.

Is there a way to get the socket client ip address when connect/disconnect?

**Comment by Totodore** [comment, marked as answer]
Hey,

As explained here : https://github.com/Totodore/socketioxide/issues/101.

It depends on your http framework. If you use axum you can get the IP with the `HttpExtension` extractor. If you use something else it depends on how client IPs are implemented.

**Reply by shishamo** [reply]
Thanks for the answer ~
Do you think about implementing a handshake feature btw?

**Reply by Totodore** [reply]
What do you mean by a handshake feature ?

**Reply by shishamo** [reply]
sorry my question was confused
in JS library i used before the socket connection information like headers and remote ip address was inside handshake object
i was just thinking how remote ip could be included socket req_parts in future
but i think you answered here already
https://github.com/Totodore/socketioxide/issues/101#issuecomment-1824811427
